src/course/impl/course-service-impl.ts

```diff
@@ -94,27 +94,28 @@ export class CourseServiceImpl implements CourseService {
     }
 
     updateContentState(request: UpdateContentStateRequest): Observable<boolean> {
-        const offlineContentStateHandler: OfflineContentStateHandler = new OfflineContentStateHandler(this.keyValueStore);
-        return new UpdateContentStateApiHandler(this.networkQueue, this.sdkConfig)
-            .handle(CourseUtil.getUpdateContentStateRequest(request))
-            .pipe(
-                map((response: { [key: string]: any }) => {
-                    if (response.hasOwnProperty(request.contentId) ||
-                        response[request.contentId] !== 'FAILED') {
-                        return true;
-                    }
-                    throw new ProcessingError('Request processing failed');
-                }),
-                catchError((error) => {
-                    return of(true);
-                }),
-                mergeMap(() => {
-                    return offlineContentStateHandler.manipulateEnrolledCoursesResponseLocally(request);
-                }),
-                mergeMap(() => {
-                    return offlineContentStateHandler.manipulateGetContentStateResponseLocally(request);
-                })
-            );
+        if (!request.target) {
+            request.target = [UpdateContentStateTarget.LOCAL, UpdateContentStateTarget.SERVER];
+        }
+
+        return defer(async () => {
+            const offlineContentStateHandler: OfflineContentStateHandler = new OfflineContentStateHandler(this.keyValueStore);
+
+            if (request.target!.indexOf(UpdateContentStateTarget.SERVER) > -1) {
+                try {
+                    await (new UpdateContentStateApiHandler(this.networkQueue, this.sdkConfig)
```

Comment:
There is a scenario where API fails to update but still gives 200 response code. What will happen in that case?